trinity/__init__.py

```diff
@@ -1,14 +1,46 @@
-import pkg_resources
+import os
+import re
+import subprocess
 import sys
 
+import pkg_resources
+
+
+def read(*parts: str) -> str:
```

Comment:
I think you can drop this function and instead do:

```python
from pathlib import Path
SETUP_FILE_PATH = Path(__file__).parent.parent / "setup_trinity"
```

Then in `find_version`:

```python
def find_version(trinity_setup_path: Path) -> str:
    version_file = trinity_setup_path.read_text()
```

We've been using `pathlib.Path` objects more and more and they are very nice to work with.